crates/astria-conductor/src/executor/mod.rs

```diff
@@ -437,8 +513,60 @@ impl Executor {
         Ok(executed_block)
     }
 
+    #[instrument(
+        skip_all,
+        fields(
+            firm_number = self.state.firm_number(),
+            soft_number = self.state.soft_number(),
+        )
+    )]
+    async fn populate_blocks_pending_finalization(
+        &mut self,
+        mut client: Client,
+    ) -> eyre::Result<()> {
+        if self.firm_blocks.is_none() || self.soft_blocks.is_none() {
+            debug!(
+                firm = self.firm_blocks.is_some(),
+                soft = self.soft_blocks.is_some(),
+                "configured without firm or soft commitments; ignoring blocks pending \
+                 finalization and not requesting them from the rollup (only activated in \
+                 soft-and-firm mode)"
+            );
+            return Ok(());
+        }
+
+        let range_of_missing_heights =
+            (self.state.firm_number().saturating_add(1))..=self.state.soft_number();
+        if range_of_missing_heights.is_empty() {
+            info!(
+                "all blocks on rollup have been finalized up to head (because firm == soft \
+                 number). Not pending blocks and continuing with normal operation."
+            );
+            return Ok(());
+        }
+        info!(
+            "rollup has not yet finalized blocks (because firm < soft number); requesting pending \
+             blocks from the rollup",
+        );
+        let blocks = client
+            .batch_get_blocks(range_of_missing_heights)
+            .await
+            .wrap_err("failed getting blocks for not yet finalized firm heights")?;
+
+        info!(
+            pending_blocks = %json(&reporting::ReportBlocks(&blocks)),
+            "received blocks pending finalization",
+        );
```

Comment:
this could be a lot of blocks? We probably don't want to have an info log with all of them? IE if variance window is 100, 600 blocks?

Reply:
You are right, this is not actionable information. I have removed the `pending_blocks` field entirely. The `info!` event now leans on the `start` and `end` block numbers that are part of the `populate_blocks_pending_finalization` span.